JSON_serializer/Tests/deserializer_test.py

```diff
@@ -0,0 +1,137 @@
+            'missing bracket [ ', 'missing bracket ] ', 'extra bracket ] C ', 'extra bracket ] D '])
+    def test_deserialize_brackets(self, t_inp, out):
+        des_inp = obj.deserialize(t_inp)
+        assert des_inp == out
+
+    @pytest.mark.parametrize('t_inp, out',
+                             [
+                                 ('{"Films about Hank Pym" :["Ant-Man", "Ant-Man and the Wasp",,"Avengers: Endgame"]}',
+                                  {'Films about Hank Pym': []}),
+                                 ('{"name":"Lord of Thunder",, "hobby":["Play with Mjoelnir and Axe", '
+                                  '"Shoot lightning from his fingers"]}', {}),
+                                 ('{"name":"Lord of Thunder", "hobby":["Play with Mjoelnir,,,, Axe", '
+                                  '"Shoot lightning from his fingers"]}', {'name': 'Lord of Thunder', 'hobby':
+                                     ['Play with Mjoelnir,,,, Axe', 'Shoot lightning from his fingers']}),
+                                 ('{"name",:"Lord of Thunder"}', {})
+                             ], ids=['extra comma between el. of list', 'extra comma between el. of dict',
+                                     'extra comma in commas', 'extra comma before colon :'])
+    def test_deserialize_comma(self, t_inp, out):
+        des_inp = obj.deserialize(t_inp)
+        assert des_inp == out
+
+    def test_json_format(self):
+        json_str = """
+                {
+                 "firstName":"Bruce",
+                 "lastName":"Banner",
+                 "isAlive":true,
+                 "birthYear":1969,
+                 "wife":null,
+                 "characteristic":[
+                   "Green",
+                   "Dual personality"
+                 ],
+                 "friends":[
+                   {
+                     "name":"Thor", 
+                     "career":"God of Thunder"
+                   },
+                   {
+                     "name":"Black Widow", 
+                     "career":"Spy"
+                   }
+                 ]
+                }
+        """
+        des_inp = obj.check_format(json_str)
+        json_out = json.loads(json_str)
+        assert des_inp == json_out
+
+    @pytest.mark.parametrize('t_inp, j_out',
+                             [
+                                 ('{"fullName": "Bruce \"Hulk\" Banner"}', {'fullName': 'Bruce "Hulk" Banner'}),
+                                 ('{"database": "HowlingCommandos\\\\Survivor\\\\Bucky \"Winter Soldier\" Barnes"}',
+                                  {'database': 'HowlingCommandos\\Survivor\\Bucky \"Winter Soldier\" Barnes'})
+                             ])
+    def test_deserialize_symbols(self, t_inp, j_out):
+        des_inp = obj.deserialize(t_inp)
+        json_out = json.loads(json.dumps(j_out))
+        assert des_inp == json_out
+
+
+if __name__ == '__main__':
```

Comment:
You don't need this. You should be able to run your tests by simple `pytest Tests`. If that's not - you should fix this

Reply:
It works with `pytest Tests`. I just thought it should be in program. OK, will remove it.

Reply:
Corrected (attempt no.3)